### PR TITLE
Modularised Magma Core.

### PR DESCRIPTION
**Pull request by GCHQDeveloper42**
Modularised Magma Core code base 

Modules: 
- `uk.gov.gchq.magmacore` - Core modules, exports MagmaCoreService for use by implementing classes.
- `uk.gov.gchq.magmacore.examples` - Provides example code and demos for using Magma Core & HQDM.

### CODE REVIEW

**Comment by GCHQDeveloper42** [comment]
The method `MagmaCoreServiceFactory.createWithJenaDatabase()` is warning:
> The type MagmaCoreJenaDatabase is not exported from this moduleJav

Because `MagmaCoreJenaDatabase` is not accessible from outside of the module. Can this method be deleted now?

File: core/src/main/java/module-info.java [hunk 1]
```diff
@@ -13,23 +13,23 @@
...
-    requires org.apache.jena.fuseki.core;
+    requires org.apache.jena.dboe.base;
     requires org.apache.jena.rdfconnection;
     requires org.apache.jena.tdb2;
     requires com.fasterxml.jackson.annotation;
 
-    requires org.apache.logging.log4j;
-    requires org.apache.commons.lang3;
+    requires uk.gov.gchq.hqdm.core;
+    requires transitive uk.gov.gchq.hqdm.rdf;
```
[the remark this answers is not in the document]
Comment (twalmsley) [reply core/src/main/java/module-info.java:29]: 👍

**Comment by twalmsley** [comment]
Yes, the factory method that takes a `MagmaCoreJenaDatabase` as a parameter can be deleted.

**Review by twalmsley — APPROVED** [review]
Looks good.

**Pull request merged**